examples/flyin_plugin/flyin_plugin/vscode.py

```diff
@@ -0,0 +1,169 @@
+#
+# For inspecting intermediate states, set breakpoints in the Python code and use the debugger for tracing.
+#
+# The `launch.json` file generated by FlyIn simply offers a convenient method to run the task. You can still use VSCode as you would locally. For instance, you can run a Python script directly from the embedded terminal using `python hello.py`.
+#
+# <img src="https://raw.githubusercontent.com/ByronHsu/static-resources/byhsu/flyin/flytesnacks/integrations/flyin/interactive_debugging.png">
+
+
+# ### 4. Resuming Task with Updated Code:
+# After you finished debugging, you can resume the task with updated code by executing the "resume task" configuration.
+# This will **terminate the code server**, run the task with inputs from the previous task, and write the output to Flyte storage.
+#
+# ** Note: Please remember to persist your code somewhere before resuming the task because you will lose the connection to vscode afterwards.**
+# <img src="https://raw.githubusercontent.com/ByronHsu/static-resources/byhsu/flyin/flytesnacks/integrations/flyin/resume_task.png">
+
+# ## Advanced Usage
+# ### 1. Installing Extensions
+# Like local VSCode, you can install a variety of extensions to assist development. Available extensions differ from official VSCode for legal reasons and are hosted on [Open VSX Registry](https://open-vsx.org/).
+#
+# Python and Jupyter extensions are installed by default. Additional extensions can be added as shown below:
+#
+# %%
+from flytekit import task, workflow
+from flytekitplugins.flyin import COPILOT_EXTENSION, VscodeConfig, vscode
+
+config = VscodeConfig()
+config.add_extensions(COPILOT_EXTENSION)  # Use predefined URL
+config.add_extensions(
+    "https://open-vsx.org/api/vscodevim/vim/1.27.0/file/vscodevim.vim-1.27.0.vsix"
+)  # Copy raw URL from Open VSX
+
+
+@task(container_image="localhost:30000/flytekit-flyin:0.0.0")
+@vscode(config=config)
+def t_ext():
+    pass
+
+
+@workflow
+def wf_ext():
+    t_ext()
+
+
+# %% [markdown]
+# ### 2. Resource Management
+# To manage GPU resources, FlyIn can terminate pods after a period of idleness (no active HTTP connections). Idleness is monitored via a heartbeat file.
+# %%
+from flytekit import task, workflow
+from flytekitplugins.flyin import vscode
+
+
+@task
+@vscode(max_idle_seconds=60000)  # 60000 seconds
+def task_with_max_idle():
+    pass
+
+
+@workflow
+def wf_idle():
+    task_with_max_idle()
+
+
+# %% [markdown]
+# ### 3. Pre/Post Execution
+#
+# For tasks requiring setup or cleanup, FlyIn allows execution of functions before and after VSCode starts.
+#
+# %%
+from flytekit import task, workflow
+from flytekitplugins.flyin import vscode
+
+
+def set_up_proxy():
+    print("set up")
+
+
+def push_code():
+    print("push code")
+
+
+@task
+@vscode(pre_execute=set_up_proxy, post_execute=push_code)
+def t_hook():
+    pass
+
+
+@workflow
+def wf_hook():
+    task_with_max_idle()
+
+
+# %% [markdown]
+# ### 4. Running Along with Task
```

Comment:
```suggestion
# ### Inspect tasks after failure
```

Reply:
Maybe change it to "Run along with task and inspect after failure". The proposed change is not precise enough

Reply:
changed to "Running along and inspect tasks after failure"

Reply:
Is FlyIn running alongside the task and initiating VSCode after task failure? If so, I suggest:

```
# ### Run FlyIn alongside tasks to initiate VSCode after failure
```